src/balena-utils.ts

```diff
@@ -111,6 +113,14 @@ export async function push(
 		pushOpt.push('--debug');
 	}
 
+	if (buildOpt.note) {
+		pushOpt.push('--note');
+
+		// sanitize note string to escape quotes
+		const note = buildOpt.note.trim().replace(/"/g, '\\"').replace(/'/g, "\\'");
```

Comment:
This will set the note to an empty string if the note is all whitespaces. I think we can probably trim before validating to avoid setting the flag for empty strings.

Reply:
core.getInput() should have already handled the trimming I believe. Will add it in to be safe